pkg/katapult/virtual_machine_group.go

```diff
@@ -1,18 +1,48 @@
 package katapult
 
 import (
+	"bytes"
 	"context"
+	"encoding/json"
 	"fmt"
 	"net/url"
 
 	"github.com/augurysys/timestamp"
 )
 
+var (
+	nullBytes               = []byte("null")
+	NullVirtualMachineGroup = &VirtualMachineGroup{null: true}
+)
+
 type VirtualMachineGroup struct {
 	ID        string               `json:"id,omitempty"`
 	Name      string               `json:"name,omitempty"`
 	Segregate bool                 `json:"segregate,omitempty"`
 	CreatedAt *timestamp.Timestamp `json:"created_at,omitempty"`
+	null      bool
+}
+
+func (s *VirtualMachineGroup) UnmarshalJSON(b []byte) error {
+	type alias VirtualMachineGroup
+
+	if bytes.Equal(b, nullBytes) {
+		s.null = true
```

Comment:
I forgot to mention we should do this:

```suggestion
		*s = VirtualMachineGroup{null: true}
```

This will replace the underlying struct that the `s` pointer refers to, with a new VM group struct which only has the `null` field populated.

That ensures that the resulting struct has nothing else set. For example, given the following marshaling:

```go
g := &VirtualMachineGroup{
	ID:        "id",
	Name:      "name",
	Segregate: true,
	CreatedAt: timestampPtr(934834834),
	null:      true,
}

json.Unmarshal([]byte("null"), g)
fmt.Printf("g: %+v\n", g)
```

Only setting `null` like we are right now, yields the following out:

```
g: &{ID:id Name:name Segregate:true CreatedAt:1999-08-16 20:20:34 +0000 UTC null:true}
```

While replacing the whole struct, yields this:

```
g: &{ID: Name: Segregate:false CreatedAt:<nil> null:true}
```